src/Relation/Binary/Properties/PartialSetoid.agda

```diff
@@ -0,0 +1,47 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Additional properties for setoids
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+open import Relation.Binary.Bundles using (PartialSetoid)
+
+module Relation.Binary.Properties.PartialSetoid
+  {a ℓ} (S : PartialSetoid a ℓ) where
+
+open import Data.Product.Base using (_,_; _×_)
+open import Relation.Binary.Definitions using (LeftTrans; RightTrans)
+open import Relation.Binary.PropositionalEquality.Core as ≡ using (_≡_)
+
+open PartialSetoid S
+
+private
+  variable
+    x y z : Carrier
+
+------------------------------------------------------------------------
+-- Proofs for partial equivalence relations
+
+trans-reflˡ : LeftTrans _≡_ _≈_
+trans-reflˡ ≡.refl p = p
+
+trans-reflʳ : RightTrans _≈_ _≡_
+trans-reflʳ p ≡.refl = p
+
+p-reflˡ : x ≈ y → x ≈ x
+p-reflˡ p = trans p (sym p)
+
+p-reflʳ : x ≈ y → y ≈ y
+p-reflʳ p = trans (sym p) p
+
+p-refl : x ≈ y → x ≈ x × y ≈ y
+p-refl p = p-reflˡ p , p-reflʳ p
+
+p-reflexiveˡ : x ≈ y → x ≡ z → x ≈ z
```

Comment:
This is `Trans _≈_ _≡_ _≈_` (and similarly below), suggesting that `reflexive` may not be quite the right name? Similarly is this not just a variant of `subst`?

Reply:
Hmmm... I see what you mean... Will ponder this for a bit!

Reply:
This is not `Trans _≈_ _≡_ _≈_` because the second `x` would be a `y` in that case.

This is an extremely weird property which basically says: if x is related to something (anything) and we have evidence of that, and x and z are identical, then we can get evidence that x and z are related.

I'm at a loss for what to name it.

Reply:
I don't know about 'weird'... these properties are characteristic of being a PER rather than a full `IsEquivalence`!?

Reply:
As to names: I could call it (something like) `conditional-reflexive`, but then we would lose that the `≡.refl` instantiation of such a lemma should be `partial-refl`... so those lemmas would also need to be renamed. And then... names too long?

Reply:
Weird likely because I generically find PERs weird.

However, your comment hits the real issue on the head: _only on its domain of definition_. We really ought to define an 'is defined' construction (a la `Trans` and `Symmetric`, etc) and use it here. Then I would have understood that you were asking for some indirect evidence of being defined. Then we'll be in a decent position to think about names.

At least now I understand this combinator's meaning (when x is defined, then equality that involve it is reflexive), and that is no longer weird at all.

Reply:
Not sure I see clearly what the `IsDefined` construction would look like, and whether it would be 'free-standing', or tied to being in a PER? 

Cf. Scott's "Identity and Existence in Intuitionistic Logic" (LNM 753) which explores variations on the theme of `E t` vs. `t = t` as expressions of '`t` is defined'/'`t` exists', but does seem tied to `_=_` being an 'equality' (Leibniz-like) relation...?